File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        Map<String, String> hashMap = new HashMap<>();

        for(int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String address = st.nextToken();
            String password = st.nextToken();
            hashMap.put(address, password);
        }

        for(int j = 0; j < M; j++) {
            bw.write(hashMap.get(br.readLine()) + "\n");
        }
        br.close();
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.kt
import java.util.*
import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val st = StringTokenizer(readLine())
    val N = st.nextToken().toInt()
    val M = st.nextToken().toInt()

    val hashmap = mutableMapOf<String, String>()
    repeat(N) {
        val input = readLine().split(" ")
        hashmap[input[0]] = input[1]
    }

    BufferedWriter(OutputStreamWriter(System.out)).run {
        repeat(M) {
            hashmap[readLine()]?.let { write("$it\n") }
        }
        flush()
        close()
    }
    close()
}